### ISSUE TITLE
`hypot` with both arguments 0 returns NaN with Unitful.jl

### ISSUE THREAD

**Issue by SebastianM-C** [JuliaLang/julia#34771]
https://github.com/JuliaLang/julia/pull/34316 introduced
```julia
hypot(x::T, y::T) where {T<:Number} = (z = y/x; abs(x) * sqrt(one(z) + z*z))
```
If `iszero(x)` and `iszero(y)`, the above returns NaN instead of `zero(T)`.

To trigger this one can use:
```
using Unitful
hypot(0u"m",0u"m")
```

One way to fix this would be
```julia
hypot(x::T, y::T) where {T<:Number} = !iszero(x) ? (z = y/x; abs(x) * sqrt(one(z) + z*z)) : abs(y)
```

If the above is ok, I can make a PR.

```julia
julia> versioninfo()
Julia Version 1.4.0-rc1.0
Commit b0c33b0cf5 (2020-01-23 17:23 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i7-7800X CPU @ 3.50GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-8.0.1 (ORCJIT, skylake)
Environment:
  JULIA_NUM_THREADS = 6
```

**Comment by giordano** [comment]
I think it was https://github.com/JuliaLang/julia/pull/33224 that removed too much code

**Comment by StefanKarpinski** [comment]
That attempt at avoiding overflow generically strikes me as a bit too clever, but 🤷‍♂

**Comment by JeffBezanson** [comment]
Yes might as well PR that change.

**Comment by JeffBezanson** [comment]
Fixed by #34817
